File: 121. Best Time to Buy and Sell Stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    var maxprice =0,minprice=prices[0];
    for(let i=0;i<prices.length;i++)
        {
            minprice=Math.min(minprice,prices[i]);
            maxprice = Math.max(maxprice,prices[i]-minprice)
        }
    return maxprice
};

File: 121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minprice = prices[0]
        maxprice = 0
        
        for x in prices:
            minprice = min(minprice,x)
            maxprice = max(maxprice,x-minprice)
            
        return maxprice
